cpp/arcticdb/util/constants.hpp

```diff
@@ -11,6 +11,14 @@ namespace arcticdb {
 
 using timestamp = int64_t;
 
+using position_t = int64_t;
+
+constexpr auto string_none = std::numeric_limits<position_t>::max();
+
+constexpr auto string_nan = std::numeric_limits<position_t>::max() - 1;
+
+constexpr auto NaT = std::numeric_limits<timestamp>::min();
```

Comment:
Note that the previous definition of `NaT` used `auto`, resulting in a `uint64_t`, whereas `timestamp` is an alias for `int64_t`. It was previously only used in `std::fill`, and so the behaviour was correct, but now we want to perform comparisons to `timestamp` values, so defining as a `timestamp` avoids unnecessary static casts in the comparison code while maintaining the correct behaviour in `std::fill`.